src/converters/BindingConverters.ts

```diff
@@ -46,7 +46,9 @@ function isBindingDirection(input: string | undefined): boolean {
 export function convertKeysToCamelCase(obj: any) {
   var output = {};
   for (var key in obj) {
-      let value = fromTypedData(obj[key]) || obj[key];
+      let valueFromDataType = fromTypedData(obj[key]);
+      // Only "undefined" will be replaced with origin object property.
+      let value = valueFromDataType === undefined ? obj[key] : valueFromDataType;
```

Comment:
Can you elaborate, with an example, on why **only** undefined gets the original value?

Reply:
added examples

Reply:
Thanks! One final question: does this need to handle the case where the input is `{ string : "{"foo": "bar"}" }`?

In other words, for expression bindings, don't we need to handle the case where the data is a JSON object? And, if so, do we need to recursively de-serialize it?

Reply:
No, we do not need this. there is a special type for "json", fromTypedData receives the interface:
```
    /** Properties of a TypedData. */
    interface ITypedData {

        /** TypedData string */
        string?: (string|null);

        /** TypedData json */
        json?: (string|null);

        /** TypedData bytes */
        bytes?: (Uint8Array|null);

        /** TypedData stream */
        stream?: (Uint8Array|null);

        /** TypedData http */
        http?: (AzureFunctionsRpcMessages.IRpcHttp|null);

        /** TypedData int */
        int?: (number|Long|null);

        /** TypedData double */
        double?: (number|null);

        /** TypedData collectionBytes */
        collectionBytes?: (AzureFunctionsRpcMessages.ICollectionBytes|null);

        /** TypedData collectionString */
        collectionString?: (AzureFunctionsRpcMessages.ICollectionString|null);

        /** TypedData collectionDouble */
        collectionDouble?: (AzureFunctionsRpcMessages.ICollectionDouble|null);

        /** TypedData collectionSint64 */
        collectionSint64?: (AzureFunctionsRpcMessages.ICollectionSInt64|null);
    }
}
```

Reply:
In that interface above, it seems to me that `json` and `string` resolve to the same declaration, so I'm unsure of how this interface detects the JSON case:

```JavaScript
        /** TypedData string */
        string?: (string|null);

        /** TypedData json */
        json?: (string|null);
```

But if you're confident that the two cases don't overlap in practice, then I would be ready to approve this PR. Just let me know if this has been tested / we have confidence in this and, if so, I'll happily approve :) Thanks!

Reply:
yes, it actually works in the same way for string and json.
I validated that for the such case ``({ string : '{"foo": "bar"}' })`` the logic returns the same result for old and new code so I am confident:
```
id: {
    foo: "bar"
}
```